### PR TITLE
API for Comment Widget - Edit

### PR DESCRIPTION
**Pull request by tasawernawaz**
[ECOM-6336](https://openedx.atlassian.net/browse/ECOM-6336)

- [x] API for edit comment
- [x] Add test for API

Please review this @waheedahmed @awais786

### CODE REVIEW

File: course_discovery/apps/publisher_comments/api/permissions.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import permissions
+
+
+class IsOwner(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of profile to view or edit it.
```
Comment (awais786) [inline course_discovery/apps/publisher_comments/api/permissions.py]: only edit

File: course_discovery/apps/publisher_comments/api/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        self.client.login(username=self.user.username, password=USER_PASSWORD)
+        response = self.client.patch(self.path, json.dumps(self.data), JSON_CONTENT_TYPE)
+
+        comment = Comments.objects.get(id=self.comment.id)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(comment.comment, self.data['comment'])
+
+    def test_update_without_permission(self):
```
Comment (awais786) [inline course_discovery/apps/publisher_comments/api/tests/test_views.py]: test_update_without_editing_permission

**Comment by awais786** [comment]
Just minor comments otherwise looks good.

**Review by awais786 — CHANGES_REQUESTED** [review]

**Review by awais786 — APPROVED** [review]

File: course_discovery/apps/publisher_comments/api/urls.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+""" Publisher comments API URLs. """
+from django.conf.urls import url
+
+from course_discovery.apps.publisher_comments.api import views
+
+urlpatterns = [
+    url(r'^(?P<pk>\d+)/edit/$', views.UpdateComment.as_view(), name='comment_edit'),
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/urls.py]: API endpoints should be RESTful. /api/comments/:id/. The HTTP verb will determine if the action.

File: course_discovery/apps/publisher_comments/api/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/__init__.py]: Remove the unnecessary white space.

**Review by clintonb — CHANGES_REQUESTED** [review]

**Comment by tasawernawaz** [comment]
@clintonb Addressed your comments, could you please review it again.

File: course_discovery/apps/publisher_comments/api/permissions.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import permissions
+
+
+class IsOwner(permissions.BasePermission):
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/permissions.py:4]: This needs to be tested.

File: course_discovery/apps/publisher_comments/api/serializers.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import serializers
+from course_discovery.apps.publisher_comments.models import Comments
+
+
+class UpdateCommentSerializer(serializers.ModelSerializer):
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/serializers.py]: `CommentsSerializer`

Comment (clintonb) [reply course_discovery/apps/publisher_comments/api/serializers.py]: This should also be tested. See the other serializer tests.

File: course_discovery/apps/publisher_comments/api/urls.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+""" Publisher comments API URLs. """
+from django.conf.urls import url
+
+from course_discovery.apps.publisher_comments.api import views
+
+urlpatterns = [
+    url(r'^comments/(?P<pk>\d+)/$', views.UpdateComment.as_view(), name='comment_edit'),
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/urls.py]: `name='comments'`

File: course_discovery/apps/publisher_comments/api/views.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+from rest_framework.generics import UpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from course_discovery.apps.publisher_comments.api.permissions import IsOwner
+
+from course_discovery.apps.publisher_comments.api.serializers import UpdateCommentSerializer
+from course_discovery.apps.publisher_comments.models import Comments
+
+
+class UpdateComment(UpdateAPIView):
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/views.py]: `UpdateCommentView`

File: course_discovery/apps/publisher_comments/api/views.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+from rest_framework.generics import UpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from course_discovery.apps.publisher_comments.api.permissions import IsOwner
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/views.py]: Local imports should be grouped together.

File: course_discovery/apps/publisher_comments/api/tests/test_views.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+from course_discovery.apps.publisher_comments.tests.factories import CommentFactory
+
+
+class UpdateCommentTests(TestCase):
+
+    def setUp(self):
+        super(UpdateCommentTests, self).setUp()
+
+        self.user = UserFactory.create()
+        self.comment = CommentFactory.create(user=self.user, comment="test comment")
```
Comment (clintonb) [inline course_discovery/apps/publisher_comments/api/tests/test_views.py]: The comment kwarg is not necessary.

**Review by clintonb — APPROVED** [review]
Address the remaining comments, and squash commits, before merging.

**Pull request merged**